azure-pipelines-rolling.yml

```diff
@@ -12,55 +12,83 @@ trigger:
-    snExe64: 'C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\sn.exe'
-    gacUtil: '$(Build.SourcesDirectory)\test\CommonAssemblies\Microsoft.VisualStudio.QualityTools.UnitTestFramework.dll'
-    ProductBinPath1: '$(Build.SourcesDirectory)\src\Microsoft.OData.Edm\bin\$(BuildConfiguration)'
-    mainDll1: 'Microsoft.OData.Edm.dll'
-    ProductBinPath2: '$(Build.SourcesDirectory)\src\Microsoft.OData.Core\bin\$(BuildConfiguration)'
-    mainDll2: 'Microsoft.OData.Core.dll'
-    ProductBinPath3: '$(Build.SourcesDirectory)\src\Microsoft.Spatial\bin\$(BuildConfiguration)'
-    mainDll3: 'Microsoft.Spatial.dll'
-    ProductBinPath4: '$(Build.SourcesDirectory)\src\Microsoft.OData.Client\bin\$(BuildConfiguration)'
-    mainDll4: 'Microsoft.OData.Client.dll'
-    ProductBinPath5: '$(Build.SourcesDirectory)\src\CodeGen\bin\$(BuildConfiguration)'
-    mainDll5: 'Microsoft.OData.Service.Design.T4.dll'
-    ProductBinPath: $(Build.SourcesDirectory)\bin\$(BuildPlatform)\$(BuildConfiguration)\Product
-  steps:
-    - template: buildandtest.yml
-    - template: credscan.yml
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    pool:
+      name: MSSecurity-1ES-Build-Agents-Pool
+      image: MSSecurity-1ES-Windows-2019
+      os: windows
+    customBuildTags:
+    - ES365AIMigrationTooling
+    stages:
+    - stage: build
+      jobs:
+      - job:  Debug	
+        variables:	
+          buildPlatform: 'Any CPU'	
+          buildConfiguration: 'Debug'	
+          skipComponentGovernanceDetection: true	
+          snExe: 'C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\sn.exe'	
+          snExe64: 'C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\sn.exe'	
+          gacUtil: '$(Build.SourcesDirectory)\test\CommonAssemblies\Microsoft.VisualStudio.QualityTools.UnitTestFramework.dll'	
+          ProductBinPath1: '$(Build.SourcesDirectory)\src\Microsoft.OData.Edm\bin\$(BuildConfiguration)'	
+          mainDll1: 'Microsoft.OData.Edm.dll'	
+          ProductBinPath2: '$(Build.SourcesDirectory)\src\Microsoft.OData.Core\bin\$(BuildConfiguration)'	
+          mainDll2: 'Microsoft.OData.Core.dll'	
+          ProductBinPath3: '$(Build.SourcesDirectory)\src\Microsoft.Spatial\bin\$(BuildConfiguration)'	
+          mainDll3: 'Microsoft.Spatial.dll'	
+          ProductBinPath4: '$(Build.SourcesDirectory)\src\Microsoft.OData.Client\bin\$(BuildConfiguration)'	
+          mainDll4: 'Microsoft.OData.Client.dll'	
+          ProductBinPath5: '$(Build.SourcesDirectory)\src\CodeGen\bin\$(BuildConfiguration)'	
+          mainDll5: 'Microsoft.OData.Service.Design.T4.dll'	
+          ProductBinPath: $(Build.SourcesDirectory)\bin\$(BuildPlatform)\$(BuildConfiguration)\Product	
+        steps:	
+          - template: /buildandtest.yml@self	
+          - template: /credscan.yml@self
+      - job:  Release

```

Comment:
The requests I raised in the `Debug` job section for nightly pipeline are relevant here as well:

- https://github.com/OData/odata.net/pull/2965/files#r1607127969
- https://github.com/OData/odata.net/pull/2965/files#r1607130100
- https://github.com/OData/odata.net/pull/2965/files#r1607132058
- https://github.com/OData/odata.net/pull/2965/files#r1607133198

Reply:
I've applied the changes here as well.